CameraXBasic/app/src/main/java/com/android/example/cameraxbasic/utils/MediaStoreUtils.kt

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright 2022 The Android Open Source Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.android.example.cameraxbasic.utils
+
+import android.content.ContentUris
+import android.content.Context
+import android.database.Cursor
+import android.net.Uri
+import android.os.Build
+import android.provider.MediaStore
+import androidx.core.net.toUri
+import java.io.File
+
+
+/**
+ * A utility class for accessing this app's photo storage.
+ *
+ * Since this app doesn't request any external storage permissions, it will only be able to access
+ * photos taken with this app. If the app is uninstalled, the photos taken with this app will stay
+ * on the device, but reinstalling the app will not give it access to photos taken with the app's
+ * previous instance. You can request further permissions to change this app's access. See this
+ * guide for more: https://developer.android.com/training/data-storage.
+ */
+class MediaStoreUtils(private val context: Context) {
+
+    // Suppress DATA index deprecation warning since we need the file location for the Glide library
+    @Suppress("DEPRECATION")
+    private val imageDataColumnIndex = MediaStore.Images.Media.DATA
```

Comment:
nit: these should remain as `private const val` -- should they be placed within a `private companion object` ?